### ISSUE TITLE
NPE  when using Standard serializer implementation for serializing {link java.util.Map} types

### ISSUE THREAD

**Issue by thomasKlug** [FasterXML/jackson-databind#1439]
Utilizing jackson-databind-2.7.5. 

There seems to be an issue surrounding the _filterId when serializing a Map. The Implementation we are setting a default filter by overriding, the JacksonAnnotationIntrospector findFilterId. A default filterId is set if one is not found. That Id may or may not be found, so we set the SimpleFilterProvider().setFailOnUnknownId to false.  However in the Implementation of the MapSerializer there is an issue within the serialize method specifically 

```
if (_filterId != null) {
                serializeFilteredFields(value, gen, provider,
                        findPropertyFilter(provider, _filterId, value), suppressableValue);
            } else if (suppressableValue != null) {
                serializeOptionalFields(value, gen, provider, suppressableValue);
            } else if (_valueSerializer != null) {
                serializeFieldsUsing(value, gen, provider, _valueSerializer);
            } else {
                serializeFields(value, gen, provider);
            }

```

Since the above code block assumes that the filter will be found, null gets passed into serializeFilteredFields resulting in an NPE. 

The above I believe could be written:

```
if (_filterId != null && findPropertyFilter(provider, _filterId, value) != null) {
                serializeFilteredFields(value, gen, provider,
                        findPropertyFilter(provider, _filterId, value), suppressableValue);
            } else if (suppressableValue != null) {
                serializeOptionalFields(value, gen, provider, suppressableValue);
            } else if (_valueSerializer != null) {
                serializeFieldsUsing(value, gen, provider, _valueSerializer);
            } else {
                serializeFields(value, gen, provider);
            }
```

**Comment by cowtowncoder** [comment]
Thanks, will fix.

**Comment by cowtowncoder** [comment]
Fixed for 2.7.9 / 2.8.5; of these latter is likely to be released sooner (and for 2.7, 2.7.9 may be the last patch).
